src/components/macro/macro-item/macro-item.component.ts

```diff
@@ -0,0 +1,208 @@
+        this.macroAction = editedAction;
+        this.editing = false;
+        this.updateView();
+        this.save.emit();
+    }
+
+    editAction() {
+        this.actionEditor.toggleEnabled(true);
+        this.editing = true;
+        this.edit.emit();
+    }
+
+    cancelEdit() {
+        this.editing = false;
+        this.cancel.emit();
+    }
+
+    deleteAction() {
+        this.delete.emit();
+    }
+
+    private updateView(): void {
+        this.title = this.macroAction.constructor.name;
+
+        if (this.macroAction instanceof DelayMacroAction) {
+            // Delay
+            this.iconName = 'clock';
+            let action: DelayMacroAction = this.macroAction as DelayMacroAction;
+            const delay = action.delay > 0 ? action.delay / 1000 : 0;
+            this.title = `Delay of ${delay}s`;
+        } else if (this.macroAction instanceof TextMacroAction) {
+            // Write text
+            let action: TextMacroAction = this.macroAction as TextMacroAction;
+            this.iconName = 'font';
+            this.title = `Write text: ${action.text}`;
+        } else if (this.macroAction instanceof KeyMacroAction) {
+            const action: KeyMacroAction = this.macroAction as KeyMacroAction;
+            this.setKeyActionContent(action);
+        } else if (this.macroAction instanceof MouseButtonMacroAction) {
+            const action: MouseButtonMacroAction = this.macroAction as MouseButtonMacroAction;
+            this.setMouseButtonActionContent(action);
+        } else if (this.macroAction instanceof MoveMouseMacroAction || this.macroAction instanceof ScrollMouseMacroAction) {
+            // Mouse moved or scrolled
+           this.setMouseMoveScrollActionContent(this.macroAction);
+        }
+        // TODO: finish for all MacroAction
+    }
+
+    private setKeyActionContent(action: KeyMacroAction) {
+        if (action.scancode === 0 && !action.modifierMask) {
+            this.title = 'Invalid keypress';
+            return;
+        }
+
+        const actionType = action.macroActionType;
+        if (actionType === macroActionType.PressKeyMacroAction || actionType === macroActionType.PressModifiersMacroAction) {
+            // Press key
+            this.iconName = 'hand-pointer';
+            this.title = 'Press key: ';
+        } else if (actionType === macroActionType.HoldKeyMacroAction || actionType === macroActionType.HoldModifiersMacroAction) {
+            // Hold key
+            this.iconName = 'hand-rock';
+            this.title = 'Hold key: ';
+        } else if (
+            actionType === macroActionType.ReleaseKeyMacroAction ||
+            actionType === macroActionType.ReleaseModifiersMacroAction
+        ) {
+            // Release key
+            this.iconName = 'hand-paper';
+            this.title = 'Release key: ';
+        }
+
+        if (action.scancode) {
+            const scancode = _find(scancodes, ['id', action.scancode.toString()]);
```

Comment:
Why do we need an additional library to solve something simple as find? [Since ES6](http://exploringjs.com/es6/ch_arrays.html), the array has a find method. Please use that instead.